packages/devtools_app/lib/src/screens/network/network_screen.dart

```diff
@@ -408,6 +412,80 @@ class MethodColumn extends ColumnData<NetworkRequest> {
   }
 }
 
+class ActionsColumn extends ColumnData<NetworkRequest>
+    implements ColumnRenderer<NetworkRequest> {
+  ActionsColumn()
+      : super(
+          '',
+          fixedWidthPx: scaleByFontFactor(32),
+          alignment: ColumnAlignment.right,
+        );
+
+  static const _actionSplashRadius = 16.0;
+
+  @override
+  bool get supportsSorting => false;
+
+  @override
+  bool get includeHeader => false;
+
+  @override
+  dynamic getValue(NetworkRequest dataObject) {
+    return '';
+  }
+
+  List<PopupMenuItem> _buildOptions(BuildContext context, NetworkRequest data) {
+    return [
+      if (data is DartIOHttpRequestData) ...[
+        PopupMenuItem(
+          child: const Text('Copy as URL'),
+          onTap: () {
+            copyToClipboard(
+              data.uri,
+              'Copied the URL to the clipboard',
+              context,
+            );
+          },
+        ),
+        PopupMenuItem(
+          child: const Text('Copy as cURL'),
+          onTap: () {
+            copyToClipboard(
+              CurlCommand.from(data).toString(),
+              'Copied the cURL command to the clipboard',
+              context,
+            );
+          },
+        )
+      ]
+    ];
+  }
+
+  @override
+  Widget build(
+    BuildContext context,
+    NetworkRequest data, {
+    bool isRowSelected = false,
+    VoidCallback? onPressed,
+  }) {
+    final options = _buildOptions(context, data);
+
+    // Only show the actions button when there are options and the row is
+    // currently selected.
+    if (options.isEmpty || !isRowSelected) return const SizedBox.shrink();
+
+    return PopupMenuButton(
+      icon: const Icon(Icons.more_vert),
+      padding: const EdgeInsets.symmetric(horizontal: densePadding),
+      splashRadius: _actionSplashRadius,
+      tooltip: '',
+      itemBuilder: (context) {
+        return options;
+      },
```

Comment:
Maybe more like `ìtemBuilder: (context) => options`? In line 475 we are doing a check if the options returned from `_buildOptions` are empty, and only display the `PopupMenuButton` when we actually have items to display.

Otherwise we would be building the options two times.

Reply:
Ah you're right. `ìtemBuilder: (context) => options` sgtm